timeseries/src/autogluon/timeseries/trainer.py

```diff
@@ -1314,20 +1232,79 @@ def refit_full(self, model: str = "all") -> Dict[str, str]:
         return copy.deepcopy(self.model_refit_map)
 
     def construct_model_templates(
-        self, hyperparameters: Union[str, Dict[str, Any]], multi_window: bool = False, **kwargs
+        self,
+        hyperparameters: Union[str, Dict[str, Any]],
+        *,
+        multi_window: bool = False,
+        path: Optional[str] = None,
+        freq: Optional[str] = None,
+        eval_metric: Optional[str] = None,
+        excluded_model_types: Optional[List[str]] = None,
+        eval_metric_seasonal_period: Optional[int] = None,
+        quantile_levels: Optional[List[float]] = None,
+        hyperparameter_tune: bool = False,
+        **kwargs,
```

Comment:
Looking at how this method is used, should we even expose `eval_metric`, `path`, `eval_metric_seasonal_period`  and `quantile_levels` as kwargs here? These attributes are set during the `__init__` of the trainer. Maybe we can just always set them to `self.*` when calling `get_preset_models`?

https://github.com/autogluon/autogluon/blob/f1bd5f42b2da0099c8d7319f38f811127446d9af/timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py#L502-L508

Reply:
agreed!